aggsender/flows/flow_aggchain_prover.go

```diff
@@ -125,6 +139,13 @@ func (a *AggchainProverFlow) GetCertificateBuildParams(ctx context.Context) (*ty
 		return nil, err
 	}
 
+	return a.verifyBuildParamsAndGenerateProof(ctx, buildParams)
+}
+
+// verifyBuildParams verifies the certificate build params and returns an error if they are not valid
+// it also calls the prover to get the aggchain proof
+func (a *AggchainProverFlow) verifyBuildParamsAndGenerateProof(
+	ctx context.Context, buildParams *types.CertificateBuildParams) (*types.CertificateBuildParams, error) {
 	if err := a.verifyBuildParams(buildParams); err != nil {
 		return nil, fmt.Errorf("aggchainProverFlow - error verifying build params: %w", err)
 	}
```

Comment:
Just a thought:
I would prefer having the `verifyBuildParams` invoked outside of this function, to have better separation of concerns (this function IMO should only be in charge of generating the proof).

Reply:
We need to do it before calling the prover, to not waste computational resources when the proof won't be generated if the data we sent was not verified first. Which can easily be done on aggsender before calling the prover.